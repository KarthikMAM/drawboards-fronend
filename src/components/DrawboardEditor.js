import React, { PureComponent } from 'react'

import Drawboard from './Drawboard'

export default class DrawboardEditor extends PureComponent {
  render () {
    if (this.props.loading) return 'loading'

    return (
      <Drawboard
        title={this.props.drawboard.title}
        overlays={this.props.drawboard.overlays}
        background={this.props.drawboard.image.key}
        onChange={this.props.onChange}
      />
    )
  }
}
