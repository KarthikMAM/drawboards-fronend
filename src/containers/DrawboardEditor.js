import React, { PureComponent } from 'react'
import { graphql, compose } from 'react-apollo'

import { getDrawboard, updateDrawboard } from '../graphql/drawboards'
import Drawboard from '../components/Drawboard'

export class DrawboardEditorContainer extends PureComponent {
  handleChange = (overlays) => this.props.updateDrawboard({
    variables: JSON.parse(JSON.stringify({
      id: this.props.data.getDrawboard.id,
      overlays: overlays,
    }, (key, value) => key === '__typename' ? undefined : value)),
    optimisticResponse: {
      __typename: 'Mutation',
      updateDrawboard: {
        ...this.props.data.getDrawboard,
        overlays: overlays,
        __typename: 'Drawboard',
      },
    },
  })

  render () {
    const { loading, getDrawboard } = this.props.data

    if (loading) return 'loading'

    return (
      <Drawboard
        title={getDrawboard.title}
        shapes={getDrawboard.overlays || []}
        background={getDrawboard.image.key}
        onChange={this.handleChange}
      />
    )
  }
}

export default compose(
  graphql(getDrawboard),
  graphql(updateDrawboard, { name: 'updateDrawboard' }),
)(DrawboardEditorContainer)
