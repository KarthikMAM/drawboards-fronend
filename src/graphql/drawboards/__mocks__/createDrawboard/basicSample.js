import { createDrawboard } from '../..'

export const createDrawboardSample1 = {
  request: {
    query: createDrawboard,
    variables: {
      title: 'Karthik',
      image: {
        bucket: 'karthik',
        key: 'hello',
      },
    },
  },
  result: {
    createDrawboard: {
      createDrawboard: {
        id: 'hello',
        title: 'welcome',
        image: {
          bucket: 'welcome',
          key: 'hi',
        },
      },
    },
  },
}
