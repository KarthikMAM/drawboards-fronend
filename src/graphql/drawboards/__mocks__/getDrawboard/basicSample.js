import { getDrawboard } from '../..'

export const getDrawboardBasicSample = {
  request: {
    query: getDrawboard,
    variables: {
      id: '1',
    },
  },
  result: {
    data: {
      getDrawboard: {
        id: '1',
        title: 'karthik',
        overlays: null,
        image: {
          bucket: 'karthik',
          key: 'karthik',
        },
      },
    },
  },
}
