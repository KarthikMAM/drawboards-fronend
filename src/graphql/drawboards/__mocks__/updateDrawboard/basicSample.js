import { updateDrawboard } from '../..'

export const updateDrawboardBasicSample = {
  request: {
    query: updateDrawboard,
    variables: {
      id: '1',
      overlays: [{
        type: 'square',
        points: [
          { x: 1, y: 0.1 },
          { x: 1, y: 0.2 },
        ],
        stroke: 'black',
        fill: 'green',
      }],
    },
  },
  result: {
    data: {
      updateDrawboard: {
        id: '1',
        title: 'Karthik',
        overlays: [{
          type: 'square',
          points: [
            { x: 1, y: 0.1 },
            { x: 1, y: 0.2 },
          ],
          stroke: 'black',
          fill: 'green',
        }],
        image: {
          bucket: 'karthik',
          key: 'karthik',
        },
      },
    },
  },
}
