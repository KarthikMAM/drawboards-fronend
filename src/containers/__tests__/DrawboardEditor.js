import React from 'react'
import { MockedProvider } from 'react-apollo/test-utils'
import { mount } from 'enzyme'

import DrawboardEditor, { DrawboardEditorContainer } from '../DrawboardEditor'
import { getDrawboardBasicSample } from '../../graphql/drawboards/__mocks__/getDrawboard/basicSample'
import { updateDrawboardBasicSample } from '../../graphql/drawboards/__mocks__/updateDrawboard/basicSample'

describe(DrawboardEditor, () => {
  it('various states', async () => { // need to be in different tests
    const wrapper = mount(
      <MockedProvider addTypename={false} cache={false} mocks={[getDrawboardBasicSample, getDrawboardBasicSample, getDrawboardBasicSample, updateDrawboardBasicSample, updateDrawboardBasicSample, updateDrawboardBasicSample]}>
        <DrawboardEditor id='1' />
      </MockedProvider>
    )

    expect(wrapper.find(DrawboardEditorContainer).props()).toMatchSnapshot('before get')

    await new Promise(resolve => setTimeout(resolve, 0))

    wrapper.update()

    expect(wrapper.find(DrawboardEditorContainer).props()).toMatchSnapshot('after get')

    wrapper.find(DrawboardEditorContainer).instance().handleChange([
      {
        type: 'square',
        points: [
          { x: 1, y: 0.1 },
          { x: 1, y: 0.2 },
        ],
        stroke: 'black',
        fill: 'green',
      },
    ])

    await new Promise(resolve => setTimeout(resolve, 0))

    expect(wrapper.find(DrawboardEditorContainer).props()).toMatchSnapshot('after update')
  })
})
