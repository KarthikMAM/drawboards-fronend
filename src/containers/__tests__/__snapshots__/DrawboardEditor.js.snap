// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL various states: after get 1`] = `
Object {
  "data": Object {
    "error": undefined,
    "fetchMore": [Function],
    "getDrawboard": Object {
      "id": "1",
      "image": Object {
        "bucket": "karthik",
        "key": "karthik",
      },
      "overlays": null,
      "title": "karthik",
    },
    "loading": false,
    "networkStatus": 7,
    "refetch": [Function],
    "startPolling": [Function],
    "stopPolling": [Function],
    "subscribeToMore": [Function],
    "updateQuery": [Function],
    "variables": Object {
      "id": "1",
    },
  },
  "id": "1",
  "updateDrawboard": [Function],
}
`;

exports[`GraphQL various states: after update 1`] = `
Object {
  "data": Object {
    "error": undefined,
    "fetchMore": [Function],
    "getDrawboard": Object {
      "id": "1",
      "image": Object {
        "bucket": "karthik",
        "key": "karthik",
      },
      "overlays": null,
      "title": "karthik",
    },
    "loading": false,
    "networkStatus": 7,
    "refetch": [Function],
    "startPolling": [Function],
    "stopPolling": [Function],
    "subscribeToMore": [Function],
    "updateQuery": [Function],
    "variables": Object {
      "id": "1",
    },
  },
  "id": "1",
  "updateDrawboard": [Function],
}
`;

exports[`GraphQL various states: before get 1`] = `
Object {
  "data": Object {
    "error": undefined,
    "fetchMore": [Function],
    "loading": true,
    "networkStatus": 1,
    "refetch": [Function],
    "startPolling": [Function],
    "stopPolling": [Function],
    "subscribeToMore": [Function],
    "updateQuery": [Function],
    "variables": Object {
      "id": "1",
    },
  },
  "id": "1",
  "updateDrawboard": [Function],
}
`;
